parquet/src/main/scala/magnolify/parquet/ParquetType.scala

```diff
@@ -102,13 +109,37 @@ object ParquetType {
     override def getWriteSupport(conf: Configuration): WriteSupport[T] = writeSupport
   }
 
+  // From AvroReadSupport
+  private val AVRO_SCHEMA_METADATA_KEY = "parquet.avro.schema"
+  private val OLD_AVRO_SCHEMA_METADATA_KEY = "avro.schema"
+
   class ReadSupport[T](private var parquetType: ParquetType[T]) extends hadoop.ReadSupport[T] {
     def this() = this(null)
 
     override def init(context: hadoop.InitContext): hadoop.ReadSupport.ReadContext = {
       if (parquetType == null) {
         parquetType = SerializationUtils.fromBase64(context.getConfiguration.get(ReadTypeKey))
       }
+
+      val metadata = context.getKeyValueMetadata
+      val model = metadata.get(ParquetWriter.OBJECT_MODEL_NAME_PROP)
+      val isAvroFile = (model != null && model.contains("avro")) ||
+        metadata.containsKey(AVRO_SCHEMA_METADATA_KEY) ||
+        metadata.containsKey(OLD_AVRO_SCHEMA_METADATA_KEY)
+      if (isAvroFile) {
+        require(
```

Comment:
These `require`s might be too strict since the read/write should still work as long as we don't touch any repeated fields. Should they be warnings instead?